server/controllers/proxyController.js

```diff
@@ -1,21 +1,48 @@
 require('dotenv').config();
+const zlib = require('zlib');
 const httpProxy = require('http-proxy');
+const cache = require('../utils/cache');
+const { rateLimiter } = require('../utils/rateLimit');
 
 const { AIRTABLE_API_KEY, AIRTABLE_ENDPOINT_URL } = process.env;
+const CONTENT_ENCODING = 'content-encoding';
+const GZIP = 'gzip';
 
 const proxy = httpProxy.createProxyServer({
   changeOrigin: true,
-  ignorePath: false
+  ignorePath: false,
+  selfHandleResponse: true,
 });
 
 proxy.on('proxyReq', function(proxyReq, req, res, options) {
   proxyReq.setHeader('authorization', `Bearer ${AIRTABLE_API_KEY}`);
 });
 
+proxy.on('proxyRes', function(proxyRes, req, res) {
+  let unzipPayload = null;
+  proxyRes.on('data', function(chunk) {
+    if (proxyRes.headers[`${CONTENT_ENCODING}`] == `${GZIP}`) {
+      zlib.gunzip(chunk, function(err, dezipped) {
+        unzipPayload = dezipped.toString();
+        cache.set(req, JSON.parse(unzipPayload));
```

Comment:
Is the `data` event for `proxyRes` guaranteed to fire only once? If not, and there are multiple chunks, we may want to wait for an `end` event to actually set this in the cache.

Reply:
Not 100% sure if it will only be fired more than once, but the `proxyRes` event is fired only once it has received a response from the Airtable server.

I tried sending the payload back only on the `end` event on `proxyRes`, this caused initial responses from the Airlock server to return a blank response (just headers from the Airlock server), but on subsequent requests (the same request) to the airlock server, the request would be fulfilled.

Reply:
@ethanlee16 the latest commit should now reflect waiting till the `end` event to parse/set cache! 🙌🏼